opm/autodiff/FlowMainEbos.hpp

```diff
@@ -410,14 +408,25 @@ namespace Opm
 
         void setupEbosSimulator()
         {
-            std::string progName("flow_ebos");
-            std::string deckFile("--ecl-deck-file-name=");
-            deckFile += param_.get<std::string>("deck_filename");
-            char* ptr[2];
-            ptr[ 0 ] = const_cast< char * > (progName.c_str());
-            ptr[ 1 ] = const_cast< char * > (deckFile.c_str());
+            std::vector<const char*> argv;
+
+            argv.push_back("flow_ebos");
+
+            std::string deckFileParam("--ecl-deck-file-name=");
+            deckFileParam += param_.get<std::string>("deck_filename");
+            argv.push_back(deckFileParam.c_str());
+
+#if defined(_OPENMP)
+            std::string numThreadsParam("--threads-per-process=");
+            int numThreads = omp_get_max_threads();
+
+            numThreadsParam += std::to_string(numThreads);
+            argv.push_back(numThreadsParam.c_str());
+#endif // defined(_OPENMP)
+
             EbosSimulator::registerParameters();
-            Ewoms::setupParameters_< TypeTag > ( 2, ptr );
+            Ewoms::setupParameters_<TypeTag>(argv.size(), &argv[0]);
+            EbosThreadManager::init();
```

Comment:
I guess this statement should be inside #if defined(_OPENMP)?

Reply:
that's not necessary, `Ewoms::ThreadManager` deals with this sort of stuff internally.